src/kiwixchoicebox.cpp

```diff
@@ -238,20 +240,25 @@ void KiwixChoiceBox::setSelections(QStringList selections, QStringList defaultSe
 
 void KiwixChoiceBox::setSelections(SelectionList selections, QStringList defaultSelection)
 {
+    clearSelections();
     choiceSelector->clear();
     for (const auto &selection: selections)
     {
         auto item = new KListWidgetItem(beautifyString(selection.second));
         item->setData(Qt::UserRole, selection.first);
         choiceSelector->addItem(item);
         if (defaultSelection.contains(selection.first)) {
-            addSelection(item);
+            defaultSelection.removeOne(selection.first);
```

Comment:
I put a comment here because it is the only line I don't know the exact purpose but it is for the whole commit.

The commit is about removing item widget if corresponding choice is not valid (switch to local library with less choices). But how do you check the choice is valid ? You seem to never check the choice against the actual library displayed.

Reply:
Yep, I went big brain on this.

`defaultSelection` will contain the current values set (in settings).
What we know is the elements in `selections` should contain all of elements in `defaultSelection`. so whenever we get that an element exists in both, we remove that element from `defaultSelections`.

In the end, if there's any element left in `defaultSelection`,  it means that isnt available in `selections` (current list of languages/categories) - it must've come from online library.

At this point, sending the signal to update choices with new values will do 2 things:
1. refresh library with our choices
2. set the correct filter in settings (that was common to `defaultSelection` and `selections`)

Reply:
> Yep, I went big brain on this.

Going big brain is a tricky way :)
If your reviewer don't understand what you do, you can be sure that you will not be able to understand it if few months. (And if you cannot, no one will do)

Why not going the basic way ?

---

In fact, I even wonder if the commit is useful.
If I have selected `wikipedia` category when browsing a remote library, I make sens to still have the filtering when a go to the local library, even if I don't have local wikipedia books.
And if we remove it, when I go back to the remote library, my filters will be changed automatically (and so I loose theme)

From a UI perspective, I would say that we clear all filters or none when we move from a library to the other.

Reply:
Thank you, I agree about the tricky way :)
I went with keeping the old list of values when changing catalog. https://github.com/kiwix/kiwix-desktop/pull/965/commits/df9873feb5dc8eaf9c45c10dc16c895a71e57292

I think another way could be that we keep the settings as a map of values like
{
English: eng
French: fra
}